Classes/Event/Indexing/BeforeDocumentsAreIndexedEvent.php

```diff
@@ -74,6 +74,15 @@ public function getDocuments(): array
     {
         return $this->documents;
     }
+    
+    /**
+     * @param Document[] $documents
+     */
+    public function setDocuments(array $documents): void
+    {
+        $this->documents = $documents;
+    }
+    
```

Comment:
setter on readonly prop?

Reply:
it is $documents, not, $documents!

Reply:
> it is $documents, not, $documents!

What do you mean?

Reply:
> > it is $documents, not, $documents!
> 
> What do you mean?

Aaah OK, you mean docoment is read only not documents.

Reply:
correct! and I need to change the documents, e.g. setting it to an empty array if it shouldn't be indexed or whatsover